### PR TITLE
Add test case for autofix of values containing comments in length-zero-no-unit

### LINKED ISSUE #4754: Fix autofix for values containing comments in length-zero-no-unit
> Clearly describe the bug

Autofix can delete comments and produce incorrect fixes when there are infix comments (comments within the content of a property value)

> Which rule, if any, is the bug related to?

Happened for me with `length-zero-no-unit`

> What code is needed to reproduce the bug?

```css
.a {
  box-shadow: -0.25rem 0rem 0rem rgba(0, 0, 0, 0), /* comment */
    inset -0.25rem 0rem 0rem rgba(0, 0, 0, 0); /* comment */
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "length-zero-no-unit": true
  }
}
```

> Which version of stylelint are you using?

`13.3.3`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `npx stylelint test.css --fix`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No, this is plain css

> What did you expect to happen?

The css above gets fixed as
```css
.a {
  box-shadow: -0.25rem 0 0 rgba(0, 0, 0, 0), /* comment */
    inset -0.25rem 0 0 rgba(0, 0, 0, 0); /* comment */
}
```

> What actually happened (e.g. what warnings or errors did you get)?

the css gets set to
```css
.a {
  box-shadow: -0.25rem 0 0 rgba(0, 0, 0, 0),
    inset -0.25rem 0000rgba(0, 0, 0, 0); /* comment */
}
```
(note the first comment is gone, and the invalid `0000rgba` is present on the third line)

### PR DESCRIPTION
**Pull request by kawaguchi1102**
> Which issue, if any, is this issue related to?

Closes #4754

> Is there anything in the PR that needs further explanation?

This issue seems to have been resolved.
Added a test to confirm normal operation.

### CODE REVIEW

File: lib/rules/length-zero-no-unit/__tests__/index.js [hunk 1]
```diff
@@ -598,6 +598,41 @@ testRule({
 			line: 1,
 			column: 25,
 		},
+		{
+			code:
+				'.a {\n' +
+				'  box-shadow: -0.25rem 0rem 0rem rgba(0, 0, 0, 0), /* comment */\n' +
+				'    inset -0.25rem 0rem 0rem rgba(0, 0, 0, 0); /* comment */\n' +
+				'}',
```
[the remark this answers is not in the document]
Comment (kawaguchi1102) [reply lib/rules/length-zero-no-unit/__tests__/index.js:606]: @ybiquitous Thanks for the review!
I'll fix it right away.

**Review by ybiquitous — APPROVED** [review]
@kawaguchi1102 Thank you. LGTM 👍🏼

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

It was likely fixed in https://github.com/stylelint/stylelint/pull/5256, when we refactored the rule to use the value parser rather than stylesearch.

I assume we should update the [13.13.0 changelog entry](https://stylelint.io/CHANGELOG#13130) rather than add this to head?

**Comment by jeddy3** [comment]
Updated existing entry to read:

> - Fixed: `length-zero-no-unit` autofix removing trailing zeroes and comments ([#5256](https://github.com/stylelint/stylelint/pull/5256)).

**Pull request merged**